### PR TITLE
prov/gni: fix local mdh for chained gets

### LINKED ISSUE #1080: gnix_rma  chained gets - local_mdh not populated
The test  -   Test(rdm_rma, read_alignment). 
Currently does a series of indirect gets and then begins doing chained gets. Once the test was edited to do a chained get before any other transaction.
```
 Test(rdm_rma, read_alignment)
 {
       //xfer_for_each_size(do_read_alignment, 1, (BUF_SZ - 1));
       do_read_buf(source, target, 62);  /* chained gets occur for len >60 */
 }
```

it bombs out with the following output

> [   170]   FMA: GNII_POST_FMA_VALIDATE_PARAMS: User Error: Invalid local memory handle
[   170]    CT: GNI_CtPostFma: CT: User Error: Entry: 0: Invalid post descriptor address
libfabric:gni:ep_data:_gnix_rma_post_req():1028<warn> [1861:1] GNI_Post*() failed: GNI_RC_INVALID_PARAM
libfabric:gni:ep_data:__gnix_vc_push_tx_reqs():2204<warn> [1861:1] Failed to push TX request 0x900fc8: Invalid argument

This occurs presumably because the `local_mdh` is not being set for chained_gets.
The tests must have been using stale data  left in the fab reqs.

@sungeunchoi @ztiffany

### LINKED ISSUE #1081: Uninitialized warning in FI_MORE path
```
prov/gni/src/gnix_rma.c: In function '_gnix_rma_more_post_req':
prov/gni/src/gnix_rma.c:800:32: warning: 'more_get' may be used uninitialized in this function [-Wmaybe-uninitialized]
  gni_ct_get_post_descriptor_t *more_get;
                                ^~~~~~~~
prov/gni/src/gnix_rma.c:799:32: warning: 'more_put' may be used uninitialized in this function [-Wmaybe-uninitialized]
  gni_ct_put_post_descriptor_t *more_put;
```

### PR DESCRIPTION
**Pull request by a-abraham**
Also fixes compiler warnings in gnix_rma.c
Fixes ofi-cray/libfabric-cray#1080
Fixes ofi-cray/libfabric-cray#1081

Signed-off-by: Amith Abraham <aabraham@cray.com>

@ztiffany @sungeunchoi

### CODE REVIEW

**Comment by ztiffany** [comment]
:+1:

File: prov/gni/src/gnix_rma.c [hunk 1]
```diff
@@ -816,11 +819,13 @@ static void __gnix_rma_more_fill_pd(struct gnix_fab_req *req,
 					 sizeof(gni_ct_put_post_descriptor_t));
 		more_put = (gni_ct_put_post_descriptor_t *)
 			   txd->gni_more_ct_descs;
+		assert(more_put);
```
Comment (sungeunchoi) [inline prov/gni/src/gnix_rma.c]: This assert (as well as the one below) would be more valuable below where you use more_get and more_put.

Comment (a-abraham) [reply prov/gni/src/gnix_rma.c]: That makes sense. Fixed.

**Comment by sungeunchoi** [comment]
👍

**Pull request merged**